src/routers/alpha-router/alpha-router.ts

```diff
@@ -1186,20 +1186,25 @@ export class AlphaRouter
     const v3Routes = cachedRoutes.routes.filter((route) => route.protocol === Protocol.V3);
     const v2Routes = cachedRoutes.routes.filter((route) => route.protocol === Protocol.V2);
     const mixedRoutes = cachedRoutes.routes.filter((route) => route.protocol === Protocol.MIXED);
-    const percents: number[] = [];
+
+    // Calculate percents from all routes, we will fetch quotes for each percent in case we had stale data when the route was cached
+    const percentsSet: Set<number> = new Set(cachedRoutes.routes.map((route) => route.percent));
```

Comment:
nit: I don't necessarily understand why we're creating a set here only to turn it into an array....I'm not necessarily a typescript wiz so maybe there's a good reason...?

Reply:
Mostly to avoid having 100 and 50 twice, so the set will handle duplicates, I opted for that rather than `if (!percents.includes(100)) percents.push(100)`